### PR TITLE
Document Pulsar sink

### LINKED ISSUE #1332: Document pulsar sink
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

Here is a doc draft for the pulsar sink.

https://www.notion.so/risingwave-labs/Sinking-Data-to-Pulsar-from-RisingWave-Draft-189290e7f38644578c6556817b228e64?pvs=4 

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Document Pulsar sink, includes new format/encode syntax 

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/12286
  https://github.com/risingwavelabs/risingwave/pull/12556

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1332
 https://github.com/risingwavelabs/risingwave-docs/issues/1350

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by emile-00** [comment]
@neverchanje 
- Is there a data type mapping table that we can add?
- The `scan.startup.mode` parameter was included in the Notion doc. Is this parameter included for Pulsar sink or is that an error?

File: docs/guides/sink-to-pulsar.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+## Parameters
+
+| Parameter Names | Description |
+| --------------- | ---------------------------------------------------------------------- |
+|topic	|Required. The address of the Pulsar topic. One source can only correspond to one topic.|
+|service.url	|Required. The address of the Pulsar service.|
+|scan.startup.mode	|Optional. The offset mode that RisingWave will use to send data. The two supported modes are earliest (earliest offset) and latest (latest offset). If not specified, the default value earliest will be used.|
+|auth.token	|Optional. A token for auth. If both `auth.token` and `oauth` are set, only `oauth` authorization is considered.|
+|oauth.issuer.url	|Conditional. The issuer url for OAuth2. This field must be filled if other `oauth` fields are specified.|
```
Comment (hengm3467) [inline docs/guides/sink-to-pulsar.md]: As a general comment, it's "URL" not "url".

**Review by neverchanje — APPROVED** [review]

**Comment by neverchanje** [comment]
- Is there a data type mapping table that we can add?
No. The way we use Pulsar is similar to how we use Kafka, where data is represented as bytes.

**Comment by neverchanje** [comment]
Sorry Emily, I forgot to validate the complete list of params. The original list was from Pulsar source.

You are right. `scan.startup.mode` is for source.

Here are some extra configs specific to Pulsar sink:

| | | 
|---|---|
| `type` | Data format. Allowed formats:<ul><li> `append-only`: Output data with insert operations.</li><li> `debezium`: Output change data capture (CDC) log in Debezium format.</li><li> `upsert`: Output data as a changelog stream. `primary_key` must be specified in this case. </li></ul> To learn about when to define the primary key if creating an `upsert` sink, see the [Overview](/data-delivery.md). |
| `max_retry_num` | Optional. The maximum number of times to retry sending a batch to Pulsar before giving up. This allows retrying in case of transient errors. It defaults to `3` |
| `retry_interval` | Optional. The time in milliseconds to wait before retrying to send a batch after a failure. It defaults to `100ms` |
| <del>`batch_size`</del> | Optional. The maximum number of messages that will be batched together and sent as a single request to Pulsar. This allows optimizing throughput by sending larger batches less frequently. It defaults to `10000` |
| <del>`batch_byte_size`</del> | Optional. The maximum size in bytes for a batch before it will be sent to Pulsar. This ensures batches will not exceed a specified size even if the message count is low. Useful for limiting network overhead and memory usage. It defaults to `1048576` |

**Comment by neverchanje** [comment]
Please do not document batch size and batch byte size for now, its behavior is still undetermined.

Rest LGTM

**Pull request merged**